shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -590,58 +592,57 @@ - (void)setDisplayingFlutterUI:(BOOL)displayingFlutterUI {
 
 - (void)callViewRenderedCallback {
   self.displayingFlutterUI = YES;
-  if (_flutterViewRenderedCallback != nil) {
-    _flutterViewRenderedCallback.get()();
-    _flutterViewRenderedCallback.reset();
+  if (self.flutterViewRenderedCallback) {
+    self.flutterViewRenderedCallback();
+    self.flutterViewRenderedCallback = nil;
   }
 }
 
 - (void)removeSplashScreenView:(dispatch_block_t _Nullable)onComplete {
-  NSAssert(_splashScreenView, @"The splash screen view must not be null");
-  UIView* splashScreen = [_splashScreenView.get() retain];
-  _splashScreenView.reset();
+  NSAssert(self.splashScreenView, @"The splash screen view must not be nil");
+  UIView* splashScreen = self.splashScreenView;
+  _splashScreenView = nil;
   [UIView animateWithDuration:0.2
       animations:^{
         splashScreen.alpha = 0;
       }
       completion:^(BOOL finished) {
         [splashScreen removeFromSuperview];
-        [splashScreen release];
         if (onComplete) {
           onComplete();
         }
       }];
 }
 
 - (void)installFirstFrameCallback {
-  if (!_engine) {
+  if (!self.engine) {
     return;
   }
 
-  fml::WeakPtr<flutter::PlatformViewIOS> weakPlatformView = [_engine.get() platformView];
+  fml::WeakPtr<flutter::PlatformViewIOS> weakPlatformView = self.engine.platformView;
   if (!weakPlatformView) {
     return;
   }
 
   // Start on the platform thread.
-  weakPlatformView->SetNextFrameCallback([weakSelf = [self getWeakNSObject],
-                                          platformTaskRunner = [_engine.get() platformTaskRunner],
-                                          rasterTaskRunner = [_engine.get() rasterTaskRunner]]() {
+  __weak FlutterViewController* weakSelf = self;
+  weakPlatformView->SetNextFrameCallback([weakSelf,
+                                          platformTaskRunner = self.engine.platformTaskRunner,
+                                          rasterTaskRunner = self.engine.rasterTaskRunner]() {
     FML_DCHECK(rasterTaskRunner->RunsTasksOnCurrentThread());
     // Get callback on raster thread and jump back to platform thread.
     platformTaskRunner->PostTask([weakSelf]() {
-      if (weakSelf) {
-        fml::scoped_nsobject<FlutterViewController> flutterViewController(
-            [(FlutterViewController*)weakSelf.get() retain]);
-        if (flutterViewController) {
-          if (flutterViewController.get()->_splashScreenView) {
-            [flutterViewController removeSplashScreenView:^{
-              [flutterViewController callViewRenderedCallback];
-            }];
-          } else {
-            [flutterViewController callViewRenderedCallback];
-          }
-        }
+      FlutterViewController* strongSelf = weakSelf;
+      if (!strongSelf) {
+        return;
+      }
+
+      if (strongSelf.splashScreenView) {
+        [strongSelf removeSplashScreenView:^{
+          [strongSelf callViewRenderedCallback];
```

Comment:
This seems to retain strongSelf during the animation. Should we check weak self again?

Reply:
Nice catch, I missed that this was a nested callback.

Two changes that would help with that to consider:
- Fix the method name to be `removeSplashScreenWithCompletion:`, as it should have been called in the first place.
- Extract the entire PostTalk lambda body to a helper method, so the whole thing is just `[weakSelf onFrameRendered]` (or whatever name), so that we don't have three nested layers of callbacks where we have to notice—which as evidenced by the fact that 3/4 of us missed it, seems like a bad thing to bank on :)

Reply:
Oh ouch -- yes! Nicely spotted.

Reply:
Done.